### DISCUSSION TITLE
Assign custom roles to specific users in Kibana

### DISCUSSION THREAD

**Discussion by Susfu** [Security-Onion-Solutions/securityonion#3778] [Unsupported Versions]
We are trying to create custom local users to give permission to read logs only and not be able to adjust any settings in kibana. We have disabled the following lines in the kibana.yaml and verified that the anonymous access is disabled. 

#xpack.security.authc.providers:
        # anonymous.anonymous1:
          # order: 0
    #    credentials: "elasticsearch_anonymous_user"
    
   We can also create users at this point and login. However if we create a role that is read only to the Default Space and also remove access to the management section it still gives them full rights. 
   
Is there anything else to change to be able to take advantage of roles fully for users?

**Comment by TOoSmOotH** [comment]
Anonymous access is enabled on Elasticsearch but I am not sure if that has anything to do with it. What are your settings for the .kibana indices?

**Reply by kiaraho** [reply]
@Susfu, we are trying to do the same thing. How did you make changes to the `/opt/so/conf/kibana/etc/kibana.yml` file persistent on the Manager node? Each time we comment out the same lines you did then run `so-kibana-restart`, Salt restores the original settings.

@TOoSmOotH, is it possible to directly login to Kibana without first logging into the SOC? For example, have analysts login via the SOC with read-only access, while admins login directly with superuser access.
